### PR TITLE
Normalise inputs

### PR DESCRIPTION
**Pull request by kgerrand**
adds ability to normalise features based on input categories. uses Elena's class InputPerVariableScaler, I just did some tidying up!

### CODE REVIEW

File: ml_baselines/modelling/train.py [hunk 1]
```diff
@@ -287,41 +361,47 @@ def train_baseline_model(site, model_type="mlp",
...
 
-    if verbose:
-        print("... predicting")
+    if normalise_inputs:
+        if verbose: print("... normalising inputs")
+        scaler = InputPerVariableScaler()
+        X_train = scaler.fit_transform(X_train)
+        X_val = scaler.transform(X_val)
+        X_test = scaler.transform(X_test)
+    else:
```
[the remark this answers is not in the document]
Comment (elenafillo) [reply ml_baselines/modelling/train.py:384]: Added tests to check it works as a standalone scaler. Adds to check it works in-pipeline to be added after merging

File: ml_baselines/modelling/train.py [hunk 2]
```diff
@@ -347,9 +427,9 @@ def train_baseline_model(site, model_type="mlp",
             "f1_val": f1_val,
             "f1_test": f1_test
         }
-        return model, X, y, scores
+        return model, X_train, y_train, scores, scaler
     else:
-        return model, X, y
+        return model, X_train, y_train, scaler
```
[the remark this answers is not in the document]
Comment (elenafillo) [reply ml_baselines/modelling/train.py]: Changed - now `train_baseline_model` returns `X, y, extra_info` where `extra_info` is a dict with labels "scores" and "scaler" (if passed `return_scores=True` and `return_scaler=True` respectively)

**Comment by elenafillo** [comment]
Looks good! I implemented Copilot suggestions, added some tests and merged main into it, resolving the conflicts. Can you take a look at the changes I've made and check that everything is running as expected for you? If so, I think it's ready to merge into main!
 
The main change is that  `train_baseline_model` now returns `model, X_train, y_train, extra_info` where `extra_info` is a dict containing `"scores"` and `"scaler"` (if `return_scores=True` and `return_scaler=True` respectively)
